### PR TITLE
Update the core to compare hash with patched digest value during build

### PR DESCRIPTION
**Pull request by kuqin12**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

The change updated build plugin to calculate hashes for MMI entry block and MM supervisor core and generate the includable dsc file for the platform/build process to consume.

This change also updated the hash verification accordingly so as to facilitate the test application operation. The aux file, hashes are communicated to supervisor in lieu of being carried by the supervisor to avoid Jormungandr loop.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [x] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [x] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [x] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

This change is tested on QEMU Q35 platform with a valid TPM interface and passed test application.

## Integration Instructions

The release process should update the usage of plugin helper functions to generate the dsc function and include the generated DSC file into platform DSC and/or release builds.

### CODE REVIEW

**Review by Javagedes — APPROVED** [review]

File: SpamPkg/Core/Runtime/StmRuntimeUtil.h [hunk 1]
```diff
@@ -79,15 +79,18 @@ IsBufferInsideMmram (
...
   this core. It will then validate the supervisor core data according to the accompanying
-  aux file and revert the executed code to the original state and measure into TPM.
+  aux file and revert the executed code to the original state and hash into TPM.
 
-  @param[in]  SpamResponderData  The pointer to the SPAM_RESPONDER_DATA structure.
-  @param[out] RetDigestList      The digest list of the image.
+  @param[in]  CpuIndex           The index of the CPU.
+  @param[in]  AuxFileBase        The base address of the auxiliary file.
+  @param[in]  AuxFileSize        The size of the auxiliary file.
+  @param[in]  MmiEntryFileSize   The size of the MMI entry file.
```
Comment (kenlautner) [inline SpamPkg/Core/Runtime/StmRuntimeUtil.h:92]: You accidentally deleted RetDigestList.

File: SpamPkg/Core/Test/ResponderValidationTestLib.c [hunk 2]
```diff
@@ -26,22 +26,23 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
...
+  to make sure the MMI entry data section is populated with legit values, then hash
   the content into TPM.
 
   The supervisor core will be verified to properly located inside the MMRAM region for
   this core. It will then validate the supervisor core data according to the accompanying
-  aux file and revert the executed code to the original state and measure into TPM.
+  aux file and revert the executed code to the original state and hash into TPM.
 
-  @param[in]  SpamResponderData  The pointer to the SPAM_RESPONDER_DATA structure.
-  @param[out] RetDigestList      The digest list of the image.
```
Comment (kenlautner) [inline SpamPkg/Core/Test/ResponderValidationTestLib.c:44]: Same as above.

File: SpamPkg/Core/Runtime/SpamResponderReport.c [hunk 3]
```diff
@@ -510,36 +180,115 @@ VerifyAndMeasureImage (
...
+  to make sure the MMI entry data section is populated with legit values, then hash
   the content into TPM.
 
   The supervisor core will be verified to properly located inside the MMRAM region for
   this core. It will then validate the supervisor core data according to the accompanying
-  aux file and revert the executed code to the original state and measure into TPM.
+  aux file and revert the executed code to the original state and hash into TPM.
 
-  @param[in]  SpamResponderData  The pointer to the SPAM_RESPONDER_DATA structure.
-  @param[out] RetDigestList      The digest list of the image.
```
Comment (kenlautner) [inline SpamPkg/Core/Runtime/SpamResponderReport.c:542]: Accidentally removed.

**Review by kenlautner — APPROVED** [review]

File: SpamPkg/Core/Runtime/SpamResponderReport.c (hunk 3 above)
Comment (kenlautner) [reply SpamPkg/Core/Runtime/SpamResponderReport.c:542]: RetDigestList specifically

Comment (kuqin12) [reply SpamPkg/Core/Runtime/SpamResponderReport.c:542]: Fixed. Thanks.

File: SpamPkg/Core/Test/ResponderValidationTestLib.c (hunk 2 above)
Comment (kuqin12) [reply SpamPkg/Core/Test/ResponderValidationTestLib.c:44]: Fixed. Thanks.

File: SpamPkg/Core/Runtime/StmRuntimeUtil.h (hunk 1 above)
Comment (kuqin12) [reply SpamPkg/Core/Runtime/StmRuntimeUtil.h:92]: Fixed. Thanks.

**Pull request merged**